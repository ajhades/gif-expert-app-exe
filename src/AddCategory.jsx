import { useState } from "react";
export const AddCategory = ({ onAddCategory }) => {
  const [name, setName] = useState("");

  const onChange = (event) => {
    setName(event.target.value);
  };

  const onSubmit = (event) => {
    event.preventDefault();
    if (name.trim().length <= 1) return;
    onAddCategory(name);
    // setCategories((categories) => [...categories, name]);
    setName("");
  };
  return (
    <form onSubmit={onSubmit}>
      <input
        type="text"
        name="name"
        id="name"
        placeholder="Nombre"
        value={name}
        onChange={onChange}
      />
    </form>
  );
};
