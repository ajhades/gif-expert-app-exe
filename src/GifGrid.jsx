import { useEffect } from "react";
import { getGifs } from "./helpers/getGifts";
export const GifGrid = ({ category }) => {
  useEffect(() => {
    getGifs(category);
  }, []);

  return (
    <article>
      <h3>{category}</h3>
      <p>{category}</p>
    </article>
  );
};
