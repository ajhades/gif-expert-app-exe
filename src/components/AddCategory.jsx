import { useState } from "react";
import PropTypes from "prop-types";
export const AddCategory = ({ onAddCategory }) => {
  const [name, setName] = useState("");

  const onChange = ({ target }) => {
    setName(target.value);
  };

  const onSubmit = (event) => {
    event.preventDefault();
    if (name.trim().length <= 1) return;
    // setCategories((categories) => [...categories, name]);
    setName("");
    onAddCategory(name);
  };
  return (
    <form onSubmit={onSubmit} aria-label="form">
      <input
        type="text"
        name="name"
        id="name"
        placeholder="Nombre"
        value={name}
        onChange={onChange}
      />
    </form>
  );
};

AddCategory.propTypes = {
  onAddCategory: PropTypes.func.isRequired,
};
