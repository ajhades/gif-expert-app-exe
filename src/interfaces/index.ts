export * from './Gif';
export * from './Categories'