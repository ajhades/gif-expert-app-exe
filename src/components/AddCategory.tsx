import React from "react";
import { useState } from "react";
import { IAddCategoryProps } from "../interfaces";

export const AddCategory = ({ onAddCategory }: IAddCategoryProps) => {
  const [name, setName] = useState<string>("");

  const onChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    setName((event.currentTarget as HTMLInputElement).value);
  };

  const onSubmit = (event: React.FormEvent<HTMLFormElement>) => {
    if (name.trim().length <= 1) return;
    
    onAddCategory(name);
    setName("");
  };

  return (
    <form onSubmit={onSubmit} aria-label="form">
      <input
        type="text"
        name="name"
        id="name"
        placeholder="Nombre"
        value={name}
        onChange={onChange}
      />
    </form>
  );
};
