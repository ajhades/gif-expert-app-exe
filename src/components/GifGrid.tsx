import { useFetchGifts } from "../hooks/useFetchGifts";
import { GifItem } from "."
import { IGifGrid } from "../interfaces";
export const GifGrid = ({ category }: IGifGrid) => {
  const { images, isLoading } = useFetchGifts(category);

  return (
    <article>
      <h3>{category}</h3>
      {isLoading && <h2>... Cargando ...</h2>}
      <ol className="card-grid">
        {images.map((image) => (
          <GifItem key={image.id} {...image} />
        ))}
      </ol>
    </article>
  );
};