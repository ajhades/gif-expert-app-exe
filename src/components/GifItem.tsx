import PropTypes, { string } from "prop-types";
import { IGifItem } from "../interfaces";
export const GifItem = ({ title, url }: IGifItem) => {
  return (
    <div className="card">
      <img src={url} alt={title} />
      <p>{title}</p>
    </div>
  );
};

GifItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};
