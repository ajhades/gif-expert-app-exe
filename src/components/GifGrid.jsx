import { useState, useEffect } from "react";
import { getGifs } from "../helpers/getGifts";
import { GifItem } from "../components/GifItem";
export const GifGrid = ({ category }) => {
  const [images, setImages] = useState([]);

  const getImages = async (category) => {
    const newImages = await getGifs(category);
    setImages(newImages);
  };
  useEffect(() => {
    getImages(category);
  }, []);

  return (
    <article>
      <h3>{category}</h3>
      <ol className="card-grid">
        {images.map((image) => (
          <GifItem key={image.id} {...image} />
        ))}
      </ol>
    </article>
  );
};
