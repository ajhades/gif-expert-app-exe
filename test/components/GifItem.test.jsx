import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components";

describe("Prueba de <GiftItem />", () => {
  const title = "HxH";
  const url = "https://media0.giphy.com/media/2y98KScHKeaQM/giphy.gif";
  test("debe hacer match con el snapshot", () => {
    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });
  test("debe mostrar la imagen con url y alt", () => {
    render(<GifItem title={title} url={url} />);

    const { src, alt } = screen.getByRole("img");
    expect(src).toBe(url);
    expect(alt).toBe(title);
  });
  test("debe mostrar el tÃ­tulo", () => {
    render(<GifItem title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy();
  });
});
