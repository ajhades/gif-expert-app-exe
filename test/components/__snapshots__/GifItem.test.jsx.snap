// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba de <GiftItem /> debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="HxH"
      src="https://media0.giphy.com/media/2y98KScHKeaQM/giphy.gif"
    />
    <p>
      HxH
    </p>
  </div>
</div>
`;
